__tests__/[slug].test.tsx

```diff
@@ -1,57 +1,43 @@
-import path from "path";
-import { promises as fs } from "fs";
-import { NextApiRequest, NextApiResponse } from "next";
-import handler from "@/pages/api/lanes/[slug]";
-
-jest.mock("fs/promises");
-
-describe("Individual swim lane API endpoints handler", () => {
-  afterEach(() => {
-    jest.restoreAllMocks();
-  });
-
-  it("should respond with 200 status and cleaned file contents for a GET request", async () => {
-    const mockReadFile = jest.spyOn(fs, "readFile");
-    mockReadFile.mockResolvedValue(JSON.stringify({ key: "value" }));
-
-    const request: NextApiRequest = {
-      method: "GET",
-      query: {
-        slug: "test-slug",
-      },
-    } as unknown as NextApiRequest;
-
-    const response: NextApiResponse = {
-      status: jest.fn(() => response),
-      json: jest.fn(),
-    } as unknown as NextApiResponse;
-
-    await handler(request, response);
-
-    expect(response.status).toHaveBeenCalledWith(200);
-
-    expect(response.json).toHaveBeenCalledWith({ key: "value" });
-
-    expect(mockReadFile).toHaveBeenCalledWith(
-      path.join(process.cwd(), "/src/data/lanes/test-slug/data.json"),
-      "utf8"
-    );
-  });
-
-  it("should respond with an error for non-GET requests", async () => {
-    const request: NextApiRequest = {
-      method: "POST",
-      query: {
-        slug: "maps", // A valid 'slug' value
-      },
-    } as unknown as NextApiRequest;
-
-    const response: NextApiResponse = {
-      status: jest.fn(() => response),
-    } as unknown as NextApiResponse;
-
-    await handler(request, response);
-
-    expect(response.status).not.toHaveBeenCalledWith(200);
-  });
+import slugHandler from "@/pages/api/lanes";
+import { NextApiRequest } from "next";
+
+it("should respond with 200 status and lanes data for a GET request", () => {
+  const request: NextApiRequest = {
+    method: "GET",
+    query: {
+      slug: "maps",
+    },
+  } as unknown as NextApiRequest;
+
+  const mockResponse: any = {
+    status: jest.fn(function () {
+      return this;
+    }),
+    json: jest.fn(),
+  };
+
+  slugHandler(request, mockResponse);
+
+  expect(mockResponse.status).toHaveBeenCalledWith(200);
+});
+
+it("should respond with an error for nonexistent slugs", () => {
+  const request: NextApiRequest = {
+    method: "GET",
+    query: {
+      slug: "not-a-slug",
+    },
+  } as unknown as NextApiRequest;
+
+  const mockResponse: any = {
+    status: jest.fn(function () {
+      return this;
+    }),
+    json: jest.fn(),
+  };
+
+  slugHandler(request, mockResponse);
+
+  expect(mockResponse.status).toHaveBeenCalledWith(404);
```

Comment:
@EdwinGuzman If you have a minute to look at this– I've been trying new approaches since Friday afternoon and cannot get this test to pass, though it enters the `else` block correctly (in `[slug].ts`) and sends the error message? Status is always 200.

Reply:
@7emansell it's because of where the `slugHandler` function is being imported from. It should be `"@/page/api/lanes/[slug]"` to get the right handler for the endpoint. It should work once that's updated.

This brings up another good point about the test that _is_ passing above. It only tests for 200 status but not if the response is only one lane. You **don't** need to make that update now.

What would be great is to structure the `__tests__` folder so that this file and `lanes.test.tsx` are under a similar structure as the structure where the file is located. So in this case `__tests__/api/lanes/...`.

While we're at it, `__tests__/index.test.tsx` is not an issue now but it could be confusing later when new files are added. Do you think that should be renamed now or at all? It can wait until later.

Reply:
Oh my god never would have noticed that! Thank you. And I can rename it to `__tests__/homepage.test.tsx` if that makes sense to you? I guess we don't have a strict test naming structure yet

Reply:
Yes, that does make sense, thanks!